libcudacxx/include/cuda/__functional/address_stability.h

```diff
@@ -60,6 +66,69 @@ proclaim_copyable_arguments(F f) -> __callable_permitting_copied_arguments<F>
   return __callable_permitting_copied_arguments<F>{_CUDA_VSTD::move(f)};
 }
 
+// Specializations for libcu++ function objects are provided here to not pull this include into `<cuda/std/...>` headers
+
+#if _CCCL_STD_VER >= 2017
+template <typename _Fn>
+struct proclaims_copyable_arguments<_CUDA_VSTD::__not_fn_t<_Fn>> : proclaims_copyable_arguments<_Fn>
+{};
+#endif // _CCCL_STD_VER > 2014
+
+template <typename _Tp>
+struct __has_builtin_operators
+    : _CUDA_VSTD::bool_constant<!_CUDA_VSTD::is_class<_Tp>::value && !_CUDA_VSTD::is_enum<_Tp>::value
+                                && !_CUDA_VSTD::is_void<_Tp>::value>
```

Comment:
We want to use variable templates when available:
```suggestion
    : _CUDA_VSTD::bool_constant<!_CCCL_TRAIT(_CUDA_VSTD::is_class, _Tp) && !_CCCL_TRAIT(_CUDA_VSTD::is_enum, _Tp)                                 && !_CCCL_TRAIT(_CUDA_VSTD::is_void, _Tp)>
```